test-case-reporting/src/test_result_record.ts

```diff
@@ -44,14 +44,26 @@ export class TestResultRecord {
     return jobId;
   }
 
-  testStatus(skipped: boolean, success: boolean): TestStatus {
+  testStatus({
+    skipped,
+    success,
+  }: {
+    skipped: boolean;
+    success: boolean;
```

Comment:
I also hate the verbosity of this syntax, but so it goes `¯\_(ツ)_/¯`

Reply:
Alas, the lengths we go for the TypeScript we love

Reply:
In that case, create a `types/relevant_module_name.d.ts` file, and in it do:
```ts
declare type ReportedTestStatus {
  skipped: boolean;
  success: boolean;
}
```

Then import it and annotate this function:
```ts
import type {ReportedTestStatus} from './types/relevant_module_name';

...

testStatus({skipped, success}: ReportedTestStatus): TestStatus {
  ...
```

It's worth the extra 3 minutes :)

Reply:
I don't think I agree with doing that -- the `{skipped, success}` structure is an artifact of the karma test reporter, and it's only used here. I also prefer `{skipped: boolean, passed: boolean}` to `ReportedTestStatus`, since the former is more immediately self-documenting.

I'm willing to change it to your suggestion if you're not persuaded, though.

Reply:
I don't feel strongly about this :)